Organization Api with Isomorphic Value Objects and Commands

rhyslbw opened the pull request:
Now we have solidified the responsibility of Space.messaging.Apis as being the validation layer, it now seems natural for us to be instantiating the aggregate targeted commands on the client.

rhyslbw committed:
Add eslintrc

rhyslbw committed:
Make commands isomorphic

I think this makes sense for commands rather than having a domain and
application scoped sets. It also makes it easy to see how we can
provide a BDD test for Api since the outcome of a CQRS Api is just to
send the command on the server-side bus.

rhyslbw committed:
Add Organization Api, move value objects and commands into shared

Validation right now is just to be logged in, a broad but minimal rule
that we _could_ run with in MVP mode.

rhyslbw committed:
Fix logical error in API, comment out user validation

I’m having trouble running this test on the client (which is to be
expected) and also setting the userId on the server to get past the
check, so have commented out for now. We may need to look at the
solution, as checking for logged in status is pretty common

rhyslbw committed:
Add donations:app to circle config

DominikGuzei commented:
Hey @rhyslbw thanks for the input! On my branch i already moved the value objects and some of the commands to the client side. I don't think that all commands should be visible on the client, as there is some internal stuff that only the server-side will handle. But i introduced the "concept" of `api-commands.js` which are the commands shared by client / server. Could also be named `api-contract`?

rhyslbw commented:
Ok np. I like naming the file _api-commands.js_. Is this currently just local?

DominikGuzei commented:
yeah, im in the middle of the registration feature, moving around some stuff :wink: but will push asap!

DominikGuzei committed:
moves value-objects and api commands to shared folder

DominikGuzei committed:
create example org in development environments

DominikGuzei committed:
updates to new process / event auto-correlation api

rhyslbw committed:
Hooks app tests in to scripts, adds debug script

DominikGuzei committed:
we have no app tests yet ;-)

rhyslbw committed:
Uses Space.getenv for devDataSetup

If the ENVs are not set this was previously throwing an error

Also updates mgp dips

rhyslbw committed:
Merge latest from feature/register
